### PR TITLE
Add dependency on boost if not using C++17

### PR DESCRIPTION
**Pull request by PhilipMourdjis**

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -33,7 +33,10 @@ def build(self):
             cmake.definitions["CMAKE_CXX_STANDARD"] = "17"
             cmake.definitions["BSONCXX_POLY_USE_STD"] = True
         else:
-            self.output.fatal("Only C++17 supported for now. Support boost dependency?")
+            cmake.definitions["CMAKE_CXX_STANDARD"] = "14"
```
Comment (promgamer) [inline conanfile.py]: Since we are using boost, should (and can) we downgrade it to C++11?

**Comment by promgamer** [comment]
Hi @PhilipMourdjis 

Thank you for the PR! :1st_place_medal: This is a much needed feature. We appreciate that you spent time on it.

**Comment by promgamer** [comment]
@PhilipMourdjis We need to add a build for non-C++17 case.

According to [the packaging tools](https://github.com/bincrafters/conan-package-tools) we can do it by going to `build.py` file and do a `build.add(...)`, where

```
add(settings=None, options=None, env_vars=None, build_requires=None): Add a new build configuration, so a new binary package will be built for the specified configuration.
```

so, we should add something like:        

```
builder.add({}, {"use_17_standard" : False}, {}, {})
```

**Review by promgamer — CHANGES_REQUESTED** [review]
We need to test it on the CI pipeline by doing something like [this](https://github.com/bisect-pt/conan-mongo-cxx-driver/pull/8#issuecomment-416922617).

File: conanfile.py (hunk 1 above)
Comment (PhilipMourdjis) [reply conanfile.py]: I think we could, it looks like it works testing locally... what would be the impact of deleting the line altogether?

Comment (promgamer) [reply conanfile.py]: I suggest we change it to `11`instead of `14`

**Comment by promgamer** [comment]
@PhilipMourdjis Did you get why the build fail?

I got it. My bad. This line is missing the `mongo-cxx-driver`. Instead of 

`builder.add({}, {"use_17_standard" : False}, {}, {})`
it should be
`builder.add({}, {"mongo-cxx-driver:use_17_standard" : False}, {}, {})`

**Comment by PhilipMourdjis** [comment]
I think there may still be problems with build.py but don't have any experience in how to set that up.

**Comment by promgamer** [comment]
Only the windows' build failed. I'm going to check that soon!

**Comment by promgamer** [comment]
```
>c:\program files (x86)\microsoft visualstudio\2017\community\vc\tools\msvc\14.15.26726\include\type_traits(1271): error C2338: You've instantiated std::aligned_storage<Len, Align> with an extended alignment (in other words, Align > alignof(max_align_t)). Before VS 2017 15.8, the member type would non-conformingly have an alignment of only alignof(max_align_t). VS 2017 15.8 was fixed to handle this correctly, but the fix inherently changes layout and breaks binary compatibility (*only* for uses of aligned_storage with extended alignments). Please define either (1) _ENABLE_EXTENDED_ALIGNED_STORAGE to acknowledge that you understand this message and that you actually want a type with an extended alignment, or (2) _DISABLE_EXTENDED_ALIGNED_STORAGE to silence this message and get the old non-conformant behavior. (compiling source file C:\Users\appveyor\.conan\data\mongo-cxx-driver\3.3.0\bisect\testing\build\639261138cd9423a9cab505939291363753290c9\source_subfolder\src\bsoncxx\builder\core.cpp) [C:\Users\appveyor\.conan\data\mongo-cxx-driver\3.3.0\bisect\testing\build\639261138cd9423a9cab505939291363753290c9\build_subfolder\source_subfolder\src\bsoncxx\bsoncxx_testing.vcxproj]
```
@pedro-alves-ferreira

**Pull request merged**